lib/spack/spack/binary_distribution.py

```diff
@@ -1798,9 +1804,22 @@ def _tar_strip_component(tar: tarfile.TarFile, prefix: str):
             continue
         m.name = m.name[result.end() :]
         if m.linkname:
```

Comment:
(summary comment) I have 3 other comments in this function. FWIW, I think they would be largely resolved by reverting to the prior logic (this comment might be more useful to consider after reading the others).